decoders/0435-owncloud_decoders.xml

```diff
@@ -53,21 +53,21 @@
 <decoder name="owncloud-failed1">
   <parent>owncloud</parent>
   <prematch>Login failed: user </prematch>
-  <regex offset="after_prematch">^'(\w+)' , wrong password, IP:(\d+.\d+.\d+.\d+)</regex>
+  <regex offset="after_prematch">^'(\w+)' , wrong password, IP:(\S+)"</regex>
   <order>user, srcip</order>
 </decoder>
 
 <decoder name="owncloud-failed2">
   <parent>owncloud</parent>
   <prematch>Login failed: </prematch>
-  <regex offset="after_prematch">^'(\w+)' \(Remote IP: '(\d+.\d+.\d+.\d+)</regex>
+  <regex offset="after_prematch">^'(\w+)' \(Remote IP: '(\S+)'</regex>
```

Comment:
```suggestion
  <regex offset="after_prematch">^'(\w+)' \(Remote IP: '(\S+)'|\)</regex>
```

The decoder.xml shows the following two different lines, seems some ownCloud versions had a different (broken) syntax:

```
(Remote IP: '127.0.0.1')",
```

and

```
(Remote IP: '127.0.0.1)",
```

See: https://github.com/ossec/ossec-hids/pull/1725

Reply:
Hi @iasdeoupxe 

I'm sorry to inform you that your solution doesn't work:
```
Sep  1 20:16:09 foo ownCloud[15463]: {core} Login failed: 'test' (Remote IP: '127.0.0.1)


**Phase 1: Completed pre-decoding.
       full event: 'Sep  1 20:16:09 foo ownCloud[15463]: {core} Login failed: 'test' (Remote IP: '127.0.0.1)'
       timestamp: 'Sep  1 20:16:09'
       hostname: 'foo'
       program_name: 'ownCloud'
       log: '{core} Login failed: 'test' (Remote IP: '127.0.0.1)'

**Phase 2: Completed decoding.
       decoder: 'owncloud'

**Phase 3: Completed filtering (rules).
       Rule id: '87301'
       Level: '6'
       Description: 'ownCloud authentication failed.'
**Alert to be generated.
```

The solution to the problem you are reporting could be as follows:
```xml
<regex offset="after_prematch">^'(\w+)' \(Remote IP: '(\S+)'|^'(\w+)' \(Remote IP: '(\S+)\)</regex>
```

Regards,
Eva